### ISSUE TITLE
Saving a model after a warm start

### ISSUE THREAD

**Issue by ragyibrahimRNA** [facebook/prophet#1701]
I am using the below code to fit a model using a warm start. which calls the below function to write the model to file.
I am able to save the initial model without any issues. However, when I come to try and save a model after a warm start (that is essentially two models that have been combined) I get the below error:

`Object of type ndarray is not JSON serializable` from the `json.dump(model_to_json(model)` code.

Can anyone help, this is quite urgent. Thank you! 

`    m_current = read_model_from_cloud(userId,
                                      containerId,
                                      tagId)
 
        m = Prophet.Prophet(daily_seasonality='auto',
                            yearly_seasonality='auto',
                            weekly_seasonality='auto',
                            interval_width=interval_width,
                            changepoint_range=0.8)
        if m_current is not None:
            m = m.fit(df, init=stan_init(m_current))
        else:
            m = m.fit(df)
        upload_model_to_cloud(m,
                              userId,
                              containerId,
                              tagId,
                              lastRun)`

`with open(model_name, 'w') as fout:
        json.dump(model_to_json(model), fout)`

The issue seems to be with the `model_to_json` function

![Screen Shot 2020-10-01 at 9 58 04 pm](https://user-images.githubusercontent.com/52903382/94806476-5885bb80-0431-11eb-96cf-55bb0ba16688.png)

**Comment by Vijay2611** [comment]
I am facing the same issue. were you able to resolve it?

**Comment by ragyibrahimRNA** [comment]
No, unfortunately no solution yet. Have you had any luck?

**Comment by bletham** [comment]
This came up recently in #1656. As discussed there, it should be the case that any numpy arrays are being converted to lists before serialization. Obviously one is being missed here but I don't know which. Is it possible to post a full set of code that reproduces the error? Or are you able to enter into a debugger and figure out which attribute it is trying to serialize when it runs into this error?

**Comment by ragyibrahimRNA** [comment]
Happy to debug... but tbh I don't exactly know what I'm looking for. Can you guide me?

**Comment by komf41** [comment]
Hi, we have the same issue here, any update on this?, what can be done to save a model after a warm start?. thanks for any advice.

**Comment by bletham** [comment]
What would be helpful is if someone could post the full set of code and data to fit a model and then hit this error when saving it to json so I can replicate the issue and see where the json conversion needs to be fixed.

**Comment by willavv** [comment]
@bletham Hi，I think the problem is caused by the fit_kwargs attribute. If this attribute is not saved, will it affect the prediction result?
![image](https://user-images.githubusercontent.com/20501232/99660175-f6465000-2a9c-11eb-8565-5c58c5133b02.png)

**Comment by ahartikainen** [comment]
Here is one solution we use in PyStan3

https://github.com/stan-dev/pystan-next/blob/74148fa9e14b2c83f3f58cb9b8271db76e876e0c/stan/model.py#L22

**Comment by Fasturtle1** [comment]
> it should be the case that any numpy arrays are being converted to lists before serialization.

Using the debugger, I found two fields "**delta**" and "**beta**" which remain in the **ndarray** format. 
Convert them to a list format manually before calling the "**model_to_json**" method.

```
model.fit_kwargs['init']['delta'] = refit_model_class.model.fit_kwargs['init']['delta'].tolist()
model.fit_kwargs['init']['beta'] = refit_model_class.model.fit_kwargs['init']['beta'].tolist()
```

**Comment by bletham** [comment]
@Fasturtle1 thanks, I see what's happening now - it's the fit_kwargs that aren't being serialized correctly. Right now those are being assumed as simple types:
https://github.com/facebook/prophet/blob/ee592451f2750e780c3e8cbaddab674cff5242a2/python/fbprophet/serialize.py#L24
but as discovered here when they are being specified for warm-starting, it can be have entry `Dict[str: np.ndarray]`.

The workaround is as you have described, but we'll have to figure out the best way to handle this properly in both serialization (which will basically be what you've done) and also in deserialization.

**Comment by bletham** [comment]
This is fixed in 29f1417.

test code:
```Python
from fbprophet.serialize import model_to_json, model_from_json
from fbprophet import Prophet
import pandas as pd


def stan_init(m):
    res = {}
    for pname in ['k', 'm', 'sigma_obs']:
        res[pname] = m.params[pname][0][0]
    for pname in ['delta', 'beta']:
        res[pname] = m.params[pname][0]
    return res

df = pd.read_csv('../examples/example_wp_log_peyton_manning.csv')
df1 = df.loc[df['ds'] < '2016-01-19', :]  # All data except the last day
m1 = Prophet().fit(df1)
m2 = Prophet().fit(df, init=stan_init(m1)) 

mj1 = model_to_json(m1)  # works
mj2 = model_to_json(m2)  # failed before, now works

```

**Comment by GabrieleSW** [comment]
I'm using fbprophet 0.7.1 and i have the same issue (TypeError: Object of type ndarray is not JSON serializable). I'm try to upgrade fbprophet library but for the system is the latest one. 

- previously i save model in a json file:

    with open(path, 'w') as fout:
         json.dump(model_to_json(model), fout) 

- after i load old model from json file and i add new training data

    prophet = Prophet(weekly_seasonality=False, daily_seasonality=True)
    for column in self.vars_to_study:
        prophet.add_regressor(column,standardize=False)
    
    with open(path, 'r') as fin:
        old_model  = model_from_json(json.load(fin))  # Load model
    
    new_model = prophet.fit(df, init=self.stan_init(old_model))

- then i try to save new model
    
    with open(path, 'w') as fout:
         json.dump(model_to_json(new_model ), fout)  # Save model 

but application crash and i have this error: TypeError: Object of type ndarray is not JSON serializable.
